lib/http/request.js

```diff
@@ -262,6 +262,16 @@ class HttpRequest extends EventEmitter {
       return;
     }
 
+    if (this.httpResponse.isInternalServerError && this.retryAttempts > 0) {
+      this.retryCount = this.retryCount + 1;
+      setTimeout(()=> {
+        this.retryAttempts = this.retryAttempts - 1;
+        this.send();
+      }, 1000);
+
+      return;
+    }
+
```

Comment:
What's the issue with the existing retry mechanism in Nightwatch? Because it should automatically retry on 500 error, right?

Reply:
current retry mechanism only retries for network errors, not internal server errors

Reply:
I think it would work for session not created as well. Earlier safari module was using selenium-webdriver HTTP module replacing that with Nightwatch HTTP module would be good enough to fix this.

Reply:
Replacing the HTTP module is not enough, because even the custom http module does not handle 500 errors.